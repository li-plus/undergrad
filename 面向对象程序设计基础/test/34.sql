create database oop;
use oop;
create table t(a date, b time); 
insert into t(a, b) values("2019-03-13", "15:47:59"); 
insert into t(a, b) values("1979-09-21", "20:16:36"); 
insert into t(a, b) values("1949-10-01", "20:15:08"); 
insert into t(a, b) values("1919-05-04", "20:15:36"); 
select * from t order by a; 
/********************** Sample Output *************************
a       b       
1919-05-04      20:15:36        
1949-10-01      20:15:08        
1979-09-21      20:16:36        
2019-03-13      15:47:59    
**************************************************************/
select * from t order by b; 
/********************** Sample Output *************************
a       b       
2019-03-13      15:47:59        
1949-10-01      20:15:08        
1919-05-04      20:15:36        
1979-09-21      20:16:36      
**************************************************************/
select a, adddate(a, 20), b, addtime(b, 30) from t order by a; 
/********************** Sample Output *************************
a       ADDDATE(a,20)   b       ADDTIME(b,30)   
1919-05-04      1919-05-24      20:15:36        20:16:06        
1949-10-01      1949-10-21      20:15:08        20:15:38        
1979-09-21      1979-10-11      20:16:36        20:17:06        
2019-03-13      2019-04-02      15:47:59        15:48:29       
**************************************************************/
drop database oop;