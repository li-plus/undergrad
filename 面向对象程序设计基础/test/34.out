a       b       
1919-05-04      20:15:36        
1949-10-01      20:15:08        
1979-09-21      20:16:36        
2019-03-13      15:47:59    
a       b       
2019-03-13      15:47:59        
1949-10-01      20:15:08        
1919-05-04      20:15:36        
1979-09-21      20:16:36      
a       ADDDATE(a,20)   b       ADDTIME(b,30)   
1919-05-04      1919-05-24      20:15:36        20:16:06        
1949-10-01      1949-10-21      20:15:08        20:15:38        
1979-09-21      1979-10-11      20:16:36        20:17:06        
2019-03-13      2019-04-02      15:47:59        15:48:29       
